src/compiler/scala/tools/nsc/classpath/DirectoryClassPath.scala

```diff
@@ -137,9 +138,10 @@ trait JFileDirectoryLookup[FileEntryType <: ClassRepresentation] extends Directo
 }
 
 object JrtClassPath {
+  private val jrtFileSystemCache = new FileBasedCache[String, FileSystem]()
```

Comment:
I forgot to also use the key for `jrtClassPathCache`

Reply:
Updated. I had to rework the caching by combining the FileSystem and ClassPath cache, otherwise you could end up with closed FileSystems when compiling on multiple threads.